### PR TITLE
[#5766] Choice options separator should be multi language

### PR DESCRIPTION
**Pull request by ceciliaavila**
Fixes # 5766
#minor

## Description
This PR updates the ChoiceOptions to support LG allowing to define different template expressions for each language.
It also adds unit tests to cover the new code.

## Specific Changes

- Updated `ChoiceInput` to get the choice options from an expression.
- Updated `ConfirmInput` to get the choice options from an expression.
- Added the new class `ChoiceOptionsSet` containing the BindAsync method to map the LG expression into a _ChoiceFactoryOptions_ object.
- Updated _ChoiceInput_ and _ConfirmInput_ schemas.
- Added new tests in `ActionTests` class:
   - Action_ChoicesWithChoiceOptions.test.dialog 
   - Action_ChoicesWithChoiceOptionsTemplate.test.dialog 
   - Action_ConfirmInputWithChoiceOptions.test.dialog
   - Action_ConfirmInputWithChoiceOptionsTemplate.test.dialog
- Added new test class `ChoiceOptionsSetTests`.

## Testing
These images show the new unit tests passing and a multilingual bot using expressions to translate the choice options separators.
image
image

### CODE REVIEW

**Comment by ceciliaavila** [comment]
CI Pipeline passing:
image

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceOptionsSet.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            var lgResult = await languageGenerator.GenerateAsync(dialogContext, template, dialogContext.State).ConfigureAwait(false);
+            if (lgResult is ChoiceFactoryOptions cs)
+            {
+                return cs;
+            }
+            else if (lgResult is string str)
+            {
+                try
+                {
+                    var jObj = (JArray)JsonConvert.DeserializeObject(str);
```
Comment (sw-joelmut) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceOptionsSet.cs:57]: Are we sure a string will always be a JArray or we need some validations?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Dialogs/Microsoft.ChoiceInput.schema [hunk 2]
```diff
@@ -170,7 +170,7 @@
                     }
                 },
                 {
-                    "$ref": "schema:#/definitions/equalsExpression"
+                    "$ref": "schema:#/definitions/stringExpression"
```
Comment (sw-joelmut) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Dialogs/Microsoft.ChoiceInput.schema:173]: Why now is a string instead of equal?

**Review by sw-joelmut — APPROVED** [review]
Looks good overall

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceOptionsSet.cs (hunk 1 above)
Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceOptionsSet.cs:57]: This is how the Bind is made in the ChoiseSet class too. If the deserialization fails, it returns null.

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs [hunk 3]
```diff
@@ -247,6 +248,27 @@ private async Task<ChoiceSet> GetChoiceSetAsync(DialogContext dc)
             }
         }
 
+        private async Task<ChoiceFactoryOptions> GetChoiceOptionsAsync(DialogContext dc, string locale)
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceInput.cs:251]: Here I followed how ChoiceSet is implemented in both ChoiceInput and ConfirmInput.

**Pull request closed without merging**